process/smartContract/scrCommon/scProcessHelper_test.go

```diff
@@ -0,0 +1,268 @@
+		args.Hasher = nil
+		sch, err := scrCommon.NewSCProcessHelper(args)
+		require.Nil(t, sch)
+		require.ErrorIs(t, err, process.ErrNilHasher)
+	})
+	t.Run("NilPubkeyConverter should err", func(t *testing.T) {
+		t.Parallel()
+
+		args := getSCProcessHelperArgs()
+		args.PubkeyConverter = nil
+		sch, err := scrCommon.NewSCProcessHelper(args)
+		require.Nil(t, sch)
+		require.ErrorIs(t, err, process.ErrNilPubkeyConverter)
+	})
+	t.Run("ValidArgs should not err", func(t *testing.T) {
+		t.Parallel()
+
+		args := getSCProcessHelperArgs()
+		sch, err := scrCommon.NewSCProcessHelper(args)
+		require.NotNil(t, sch)
+		require.NoError(t, err)
+	})
+}
+
+func TestSCProcessHelper_GetAccountFromAddress(t *testing.T) {
+	t.Parallel()
+
+	t.Run("not same shard should not err", func(t *testing.T) {
+		t.Parallel()
+
+		args := getSCProcessHelperArgs()
+		args.ShardCoordinator = &testscommon.ShardsCoordinatorMock{
+			ComputeIdCalled: func(address []byte) uint32 {
+				return 1
+			},
+			SelfIDCalled: func() uint32 {
+				return 2
+			},
+		}
+		sch, _ := scrCommon.NewSCProcessHelper(args)
+
+		account, err := sch.GetAccountFromAddress([]byte{})
+		require.Nil(t, account)
+		require.Nil(t, err)
+	})
+	t.Run("load account with err should err", func(t *testing.T) {
+		t.Parallel()
+
+		expectedErr := errors.New("expected")
+
+		args := getSCProcessHelperArgs()
+		args.Accounts = &testState.AccountsStub{
+			LoadAccountCalled: func(address []byte) (vmcommon.AccountHandler, error) {
+				return nil, expectedErr
+			},
+		}
+		sch, _ := scrCommon.NewSCProcessHelper(args)
+
+		account, err := sch.GetAccountFromAddress([]byte{})
+		require.Nil(t, account)
+		require.ErrorIs(t, err, expectedErr)
+	})
+	t.Run("load account with wrong type should err", func(t *testing.T) {
+		t.Parallel()
+
+		args := getSCProcessHelperArgs()
+		args.Accounts = &testState.AccountsStub{
+			LoadAccountCalled: func(address []byte) (vmcommon.AccountHandler, error) {
+				return &testState.BaseAccountMock{}, nil
+			},
+		}
+		sch, _ := scrCommon.NewSCProcessHelper(args)
+
+		account, err := sch.GetAccountFromAddress([]byte{})
+		require.Nil(t, account)
+		require.ErrorIs(t, err, process.ErrWrongTypeAssertion)
+	})
+	t.Run("load account with success should not err", func(t *testing.T) {
+		t.Parallel()
+
+		args := getSCProcessHelperArgs()
+		args.Accounts = &testState.AccountsStub{
+			LoadAccountCalled: func(address []byte) (vmcommon.AccountHandler, error) {
+				return &testState.AccountWrapMock{}, nil
+			},
+		}
+		sch, _ := scrCommon.NewSCProcessHelper(args)
+
+		account, err := sch.GetAccountFromAddress([]byte{})
+		require.NotNil(t, account)
+		require.NoError(t, err)
+	})
+}
+
+func TestSCProcessHelper_CheckSCRBeforeProcessing(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil SCR should not err", func(t *testing.T) {
+		args := getSCProcessHelperArgs()
+		sch, _ := scrCommon.NewSCProcessHelper(args)
+
+		account, err := sch.CheckSCRBeforeProcessing(nil)
```

Comment:
Here and everywhere below, the result of this func is not an `account`

Reply:
fixed, renamed to scrData